module/data/other/character-generation.mjs

```diff
@@ -263,6 +263,35 @@ export default class CharacterGenerationData extends SparseDataModel {
     } );
   }
 
+  get namegiverAbilities() {
+    return this.abilities.namegiver;
+  }
+  
+  set namegiverAbilities( namegiverDocument ) {
+    // Only update data if namegiver changes
+    if ( !namegiverDocument || ( this.selectedNamegiver === namegiverDocument.uuid ) ) return;
+    console.log( "namegiverAbilities", namegiverDocument.system.abilities );
+  
+    this.updateSource( {
+      abilities: {
+        namegiver: Object.fromEntries( namegiverDocument.system.abilities.map( ability => [ ability, 0 ] ) ),
+      }
+    } );
+  }
+  
+  async getNamegiverAbilities() {
+    const namegiver = await this.namegiverDocument;
+    if ( !namegiver ) return [];
+  
+    const abilitiesUuid = namegiver.system.abilities || [];
+    const abilities = await Promise.all( abilitiesUuid.map( async ( abilityUuid ) => {
+      const ability = await fromUuid( abilityUuid );
+      return ability.type === "talent" ? abilityUuid : null;
```

Comment:
This only shows talents. But things like "Heat Sight" will be `specialAbility`. So it should probably check if its `system` data model is an instance of `AbilityTemplate`?